app/code/Magento/Bundle/Model/Product/LinksList.php

```diff
@@ -19,52 +25,62 @@ class LinksList
     protected $type;
 
     /**
-     * @var \Magento\Framework\Api\DataObjectHelper
+     * @var DataObjectHelper
      */
     protected $dataObjectHelper;
 
     /**
-     * @param \Magento\Bundle\Api\Data\LinkInterfaceFactory $linkFactory
+     * @var Link
+     */
+    protected $linkHelper;
+
+    /**
+     * @param LinkInterfaceFactory $linkFactory
      * @param Type $type
-     * @param \Magento\Framework\Api\DataObjectHelper $dataObjectHelper
+     * @param DataObjectHelper $dataObjectHelper
+     * @param Link $linkHelper
      */
     public function __construct(
-        \Magento\Bundle\Api\Data\LinkInterfaceFactory $linkFactory,
-        \Magento\Bundle\Model\Product\Type $type,
-        \Magento\Framework\Api\DataObjectHelper $dataObjectHelper
+        LinkInterfaceFactory $linkFactory,
+        Type $type,
+        DataObjectHelper $dataObjectHelper,
+        Link  $linkHelper
```

Comment:
I suggest to make it nullable to prevent backward-incompatibility in inherited classes.
```suggestion
        ?Link  $linkHelper = null
```

Reply:
@swnsma  as much as I understand the need for backward-incompatibility thing, i am not a fan for this thing. Can misslead developers :))) but OK.

Also, this change is not part of the fix and i do not know how it got here

Reply:
Faced with tests failing with it. Need to investigate this moment.

Reply:
Hi @sheepfy,
Yeah, this changes were done during tests implementation :)

Reply:
Hi @engcom-Hotel,
The same as I have mentioned before, please check that in case of null, object will be created via ObjectManager (e.g. \Magento\Framework\App\DocRootLocator)